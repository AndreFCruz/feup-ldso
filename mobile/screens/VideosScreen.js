import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'native-base';
import { connect } from 'react-redux';
import { getVideos } from '../reducers/modules/facultyReducer';

class VideosScreen extends React.Component {

    componentDidMount() {
        this.props.getVideos(this.props.name);
    }

    render() {
        const { name, loading } = this.props;

        if (loading) {
            return (
                <View>
                    <Text>
            Loading...
                    </Text>
                </View>
            );
        }
        return (
            <View>
                <Text>
          Placeholder for Videos Screen of {name}
                </Text>
            </View>
        );
    }
}

VideosScreen.propTypes = {
    name: PropTypes.string,
    loading: PropTypes.bool,
    getVideos: PropTypes.func
};

const mapStateToProps = ({ faculty }) => ({
    name: faculty.name,
    loading: faculty.loading,
    videos: faculty.videos
});

const mapDispatchToProps = {
    getVideos
};

export default connect(mapStateToProps, mapDispatchToProps)(VideosScreen);
