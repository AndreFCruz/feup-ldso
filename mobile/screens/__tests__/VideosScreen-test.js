import 'react-native';
import React from 'react';
import VideosScreen from '../VideosScreen';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

function setup(loading = false) {

    const initialState = { 
        faculty: {
            name: 'feup',
            loading: loading,
            videos: [
                'https://youtube.com/1',
                'https://youtube.com/2',
                'https://youtube.com/3',
            ] 
        },
        language: {
            selection: 'en'
        }
    };

    const wrapper = shallow(<VideosScreen />,
        {
            context: { store: mockStore(initialState) },
            disableLifecycleMethods: false
        });

    return wrapper;
}

it('renders loading screen', () => {
    const wrapper = setup(true);

    expect(wrapper.dive()).toMatchSnapshot();
});
    
it('renders screen with video content', () => {
    const wrapper = setup();
    jest.unmock('ScrollView');
    expect(wrapper.dive()).toMatchSnapshot();
});