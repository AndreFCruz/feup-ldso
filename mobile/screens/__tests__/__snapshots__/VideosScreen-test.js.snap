// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading screen 1`] = `<ActivityIndicatorView />`;

exports[`renders screen with video content 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "maxHeight": 1334,
      "width": 750,
    }
  }
>
  <Carousel
    name="feup"
    videos={
      Array [
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
          virtual={undefined}
        >
          <WebView
            domStorageEnabled={true}
            javaScriptEnabled={true}
            javaScriptEnabledAndroid={true}
            onNavigationStateChange={[Function]}
            scalesPageToFit={true}
            source={
              Object {
                "uri": "https://youtube.com/1?playlist=1&autoplay=1&loop=1&theme=dark&fs=0&rel=0&disablekb=1",
              }
            }
            startInLoadingState={true}
            style={
              Object {
                "elevation": 10,
                "flex": 1,
                "width": 750,
              }
            }
          />
        </Styled(ViewNB)>,
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
          virtual={undefined}
        >
          <WebView
            domStorageEnabled={true}
            javaScriptEnabled={true}
            javaScriptEnabledAndroid={true}
            onNavigationStateChange={[Function]}
            scalesPageToFit={true}
            source={
              Object {
                "uri": "https://youtube.com/2?playlist=2&autoplay=1&loop=1&theme=dark&fs=0&rel=0&disablekb=1",
              }
            }
            startInLoadingState={true}
            style={
              Object {
                "elevation": 10,
                "flex": 1,
                "width": 750,
              }
            }
          />
        </Styled(ViewNB)>,
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
          virtual={undefined}
        >
          <WebView
            domStorageEnabled={true}
            javaScriptEnabled={true}
            javaScriptEnabledAndroid={true}
            onNavigationStateChange={[Function]}
            scalesPageToFit={true}
            source={
              Object {
                "uri": "https://youtube.com/3?playlist=3&autoplay=1&loop=1&theme=dark&fs=0&rel=0&disablekb=1",
              }
            }
            startInLoadingState={true}
            style={
              Object {
                "elevation": 10,
                "flex": 1,
                "width": 750,
              }
            }
          />
        </Styled(ViewNB)>,
      ]
    }
  />
</Styled(ViewNB)>
`;
